{% extends "layouts/base.jinja2" %}

{% block head %}
  {{ super() }}
  <script type="text/javascript">
    $(document).ready(function () {
      // Every 5 seconds, refresh the metrics form the server.
      window.setInterval(refreshstats, 5000);
      refreshstats();
    });
  </script>
{% endblock %}

{% block content %}
  <div id="snapshot-template" style="display: none">
    <div class="snapshot-ui">
      <div class="row" style="margin-bottom: 10px; margin-top: 10px;">
        <div class="col s1 valign-wrapper"><i class="large material-icons" style="font-size: 30px;">archive</i></div>
        <div class="col s11">
          <div>
            <span id="name" class="left truncate"></span>
            <span class="right" style="font-weight: bold;">
              <span id="status"></span>
              <i class="material-icons small tooltipped" id="status-help" style="font-size: 15px; vertical-align: top"
                data-position="left" data-tooltip="error">help</i>
            </span>
            <br />
            <span class="left">Size: <span id="size"></span></span>
            <br />
            <a class='dropdown-trigger btn right ha-blue waves-effect' href='#' data-target='action_dropdown'
              id="action_dropdown_button" style="display: none;">
              <i class="material-icons medium left">keyboard_arrow_down</i>
              Actions
            </a>
            <br />
            <a href="#" onclick="showDetails(this);return false;" id="status-details"><i class="material-icons left"
                style="margin-right: 3px; font-size: 15px;">info</i></a>
            <i class="material-icons left icon-protected tooltipped" data-position="right"
              data-tooltip="This snapshot is password protected" style="margin-right: 3px; font-size: 15px;">vpn_key</i>
            <i class="material-icons left icon-retain tooltipped" data-position="right"
              data-tooltip="This snapshot will be kept indefinitely until you delete it."
              style="margin-right: 3px; font-size: 15px;">lock</i>
            <i class="material-icons left tooltipped icon-warn-delete yellow-text text-darken-4" data-position="right"
              data-tooltip="" style="margin-right: 3px; font-size: 15px;">warning</i>
          </div>
          <ul id='action_dropdown' class='dropdown-content'>
            <li id="delete_option">
              <a href="#!" id="delete_link" data-target="deletemodal" class="modal-trigger tooltipped"
                data-position="left" data-tooltip="Deletes the snapshot from Home Assistant, Google Drive, or both."
                onClick="confirmDeleteSnapshot(this);return false;">
                <i class="material-icons left" style="font-size: 25px">delete_forever</i>Delete
              </a>
            </li>
            <li id="download_option">
              <a href="#!" id="download_link" class="tooltipped" data-position="left"
                data-tooltip="Save the snapshot archive to your computer."
                onClick="downloadSnapshot(this);return false;">
                <i class="material-icons left" style="font-size: 25px">file_download</i>Download
              </a>
            </li>
            <li class="divider" tabindex="-1"></li>
            <li id="restore_option">
              <a href="#!" onclick="restoreClick(this)" id="restore_link" class="tooltipped" data-position="left"
                data-tooltip="Restore the snapshot, overwriting your current configuration with this archive's.">
                <i class="material-icons left" style="font-size: 25px">input</i>Restore
              </a>
            </li>
            <li id="upload_option">
              <a href="#!" id="upload_link" data-target="uploadmodal" class="modal-trigger tooltipped"
                data-position="left"
                data-tooltip="Upload the snapshot from Google Drive to Home Assistant.  <br/>You'll need to do this if you want to restore the snapshot."
                onClick="uploadSnapshot(this);return false;">
                <i class="material-icons left" style="font-size: 25px">file_upload</i>Upload
              </a>
            </li>
            <li id="retain_option">
              <a href="#!" id="retain_link" class="tooltipped" data-position="left"
                data-tooltip="Keep this snapshot indefinitely until you manually delete it."
                onClick="retainSnapshot(this);return false;">
                <i class="material-icons left" style="font-size: 25px">lock</i>Never Delete
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div class="divider">
      </div>
    </div>
  </div>
  <div class='error' style='display:none' id="toast">You shouldn't see this</div>
  <main>
    <div class="container">
      <div class="section">
        <div class="row default-hidden" id="error_card">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">sync_problem</i>
                  <span style="vertical-align: top;">Error while backing up</span>
                </span>
                <p>The add-on got the error below while trying to sync your snapshots. Some transient errors are
                  expected
                  (eg an upload could fail if the internet is down) but if this persist for more than a few minutes your
                  snapshots won't get backed up and the information you see on this page may be stale. If you file an
                  issue on GitHub with the link below, it could be helpful to provide the logs downloaded from the
                  "Download Logs" link below</p>
                <br />
                <p id="generic_error_title" style="white-space: pre-wrap;font-family: monospace;"></p>
                <div>
                  <a onclick="toggleLinkSlide(this, 'generic_error_details');" href="#!"><i class="material-icons"
                      style="padding-right: 10px; vertical-align: bottom;">arrow_drop_down</i> View Exception
                    Details</a>
                  <div id="generic_error_details" class="default-hidden"
                    style="padding-left: 10px; white-space: pre-wrap;font-family: monospace;">
                    <p>Below is the list of names and IP Addresses Google's API servers resolve to. Next to each is
                      the result from an attempt the add-on has made to ping the address. This may help in
                      determining if something in your network is blocking accesss to one of Google's API servers,
                      or if Google's API servers are having problems.</p>
                  </div>
                </div>
                <div class="card-action white-text">
                  <a id="error_github_search" href="#" target="_blank">Find similar issues</a>
                </div>
                <div class="card-action white-text">
                  <a id="error_github_link" onclick="bugReport()">File this issue on Github</a>
                </div>
                <div class="card-action white-text">
                  <a id="download_logs_link" href="log?format=download" target="_blank">Download logs</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card low_space" id="low_space">
          <div class="col s12 m9">
            <div class="card red darken-4 white-text">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">disc_full</i>
                  <span style="vertical-align: top;">Low Disc Space</span>
                </span>
                <p> It looks like you're low on disk space. The addon won't attempt to create a snapshot until you take
                  action or more space becomes available. It looks like your backup folder in Home Assistant is <b
                    id="data_pct_used"></b> full and only has <b id="data_space_remaining"></b> left so creating 
                    another snapshot runs the risk of exhausting the space available.  When Home Assistant runs out of 
                    disk space, it can stop responding and become very difficult to fix.  This addon can only estimate 
                    how big the the next snapshot will be based on your most recent one, it can't know how big the snapshot 
                    will be until its actually created.</p>
                  <p>To resolve this you can:</p>
                  <ul class="browser-default">
                    <li>Free up space in Home Assistant.  Addons like the <a target="_blank" rel="noreferrer" href="https://www.home-assistant.io/addons/configurator/">Configurator</a> can help you see what files might be taking up space.</li>
                    <li>Make your snapshots smaller (see <a target="_blank" rel="noreferrer"
                      href="https://github.com/sabeechen/hassio-google-drive-backup#will-this-fill-up-my-google-drive--why-are-my-snapshots-so-big">this
                      tip</a>).  You can also click the little <i class="material-icons"
                      style="display: inline; margin-right: 2px; vertical-align: middle; font-size: 15px">info</i> next to snapshots below to see which parts of them are taking up the most space.</li>
                      <li>Modify your <a href="#!" onclick="loadSettings()"><i class="material-icons"
                        style="display: inline; margin-right: 2px; vertical-align: middle; font-size: 15px">settings</i>add-on
                      settings</a> to store fewer snapshots in Home Assistant.</li>
                      <li>Delete existing snapshots from Home Assistant by clicking "Action" -> "Delete" on snapshots below.  Make sure you don't need them anymore or they're backed to Google Drive.</li>
                    <li>Modify your <a href="#!" onclick="loadSettings()"><i class="material-icons"
                      style="display: inline; margin-right: 2px; vertical-align: middle; font-size: 15px">settings</i>add-on
                      settings</a> and disable "Stop on low disc space".  This will make the addon stop checking disc usage, do so at your own peril.</li>
                    <li>Click "Create snapshot anyway" to bypass this warning just this one time.  This is useful if you're confident that the snapshot actually will fit in the available space.  Again, do so at your own peril.</li>
                  </ul>
                  <div class="card-action white-text">
                      <a href="#" onclick="sync('low_space'); return false;">Try Syncing Again</a>
                      <a href="#" onclick="skipLowSpaceWarning(); return false;">Create snapshot anyway</a>
                  </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card drive_full" id="drive_full_card">
          <div class="col s12 m9">
            <div class="card red darken-4 white-text">
              <div class="card-content white-text">

                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">disc_full</i>
                  <span style="vertical-align: top;">Google Drive is Full</span>
                </span>

                <p>Google Drive is full, backups can't continue until space is available. You can go to Google Drive's
                  <a href="https://drive.google.com/drive/quota" target="_blank" rel="noreferrer">quota page</a> to see whats using up space in your
                  Google
                  Drive, but because that space is shared between several Google products (eg GMail, Google Photos) one
                  of
                  those products might be using the space instead. You can see your usage across all those products on
                  Google's <a href="https://drive.google.com/settings/storage" target="_blank" rel="noreferrer">storage settings</a> page.</p>
                <p>To resolve this you can:</p>
                <ul class="browser-default">
                  <li>Free up space in Google Drive, GMail, or Google Photos by deleting things.</li>
                  <li>Modify your <a href="#!" onclick="loadSettings()"><i class="material-icons"
                        style="display: inline; margin-right: 2px; vertical-align: middle; font-size: 15px">settings</i>add-on
                      settings</a> to store fewer snapshots in Google Drive</li>
                  <li><a href="https://drive.google.com/settings/storage" target="_blank" rel="noreferrer">Buy more storage</a> from Google</li>
                  <li>Make your snapshots smaller (see <a
                      href="https://github.com/sabeechen/hassio-google-drive-backup#will-this-fill-up-my-google-drive--why-are-my-snapshots-so-big" target="_blank" rel="noreferrer">this
                      tip</a>)</li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card creds_bad" id="creds_bad_card">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">error</i>
                  <span style="vertical-align: top;">Couldn't Log in with Google</span>
                </span>
                <p>Google Drive rejected your credentials. Your snapshots will not be backed up until this is resolved.
                </p>
                <p>This can happen if:</p>
                <ul class="browser-default">
                  <li>You revoked the add-on's access in your Google account settings.</li>
                  <li>You recently changed your Google account password.</li>
                  <li>You haven't used your Google account for over 6 months.</li>
                  <li>The add-on's credentials changed, or Google changed them.</li>
                </ul>
                <p>To resolve this, you'll need to <a href="reauthenticate">authenticate</a> this add-on with Google
                  again. If that still doesn't help, you may need to update the add-on</p>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card delete_error" id="delete_error">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">delete_forever</i>
                  <span style="vertical-align: top;">Couldn't delete old snapshots in Home Assistant</span>
                </span>
                <p>
                  The add-on ran into an error (HTTP 400) trying to delete your oldest snapshot. This can happen
                  sometimes (especially if you just installed the add-on) when Home Assistant's cache of your existing snapshots
                  is out of sync. The add-on can't continue until this is resolved.
                </p>
                <p>To resolve this, you can try:</p>
                <ul class="browser-default">
                  <li>Refreshing your snapshots by going to the snapshot page in Home Assistant and clicking the refresh icon
                    in the top right.</li>
                  <li>Restarting the machine Home Assistant runs on.</li>
                </ul>
                <p>The add-on will retry periodically to check if the issue is resolved. To retry now, click "Try
                  Syncing Again" below. If the above doesn't help, please file an issue on <a onclick="bugReport()">GitHub</a>
                  to get more help.</p>
              </div>
              <div class="card-action white-text">
                <a href="#" onclick="sync('delete_error'); return false;">Try Syncing Again</a>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card google_dns" id="cant_reach_google">
          <div class="col s12 m12 l10">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">cloud_off</i>
                  <span style="vertical-align: top;">Couldn't Reach Google Drive (DNS error)</span>
                </span>
                <p>The add-on couldn't resolve Google Drive's server address. Snapshots can't be backed up
                  without being able to reach Google Drive.</p>
                <p>Please ensure:</p>
                <ul class="browser-default">
                  <li>The internet is working.</li>
                  <li>The server www.googleapis.com is reachable from your Home Assistant
                    machine. Some ad-blocking and firewall software (eg Ad Guard, Pihole, etc) can block this
                    addresses.<br />
                    <div style='padding: 5px'>
                      <a onclick="toggleLinkSlide(this, 'cant_reach_google_dns');" href="#!"><i class="material-icons"
                          style="padding-right: 10px; vertical-align: bottom;">arrow_drop_down</i> Show DNS info</a>
                      <div id="cant_reach_google_dns" class="default-hidden" style="padding-left: 10px">
                        <p>Below is the list of names and IP Addresses Google's API servers resolve to. Next to each is
                          the result from an attempt the add-on has made to ping the address. This may help in
                          determining if something in your network is blocking accesss to one of Google's API servers,
                          or if Google's API servers are having problems.</p>
                        <div class="dns_info row"></div>
                        <div class="divider"></div>
                      </div>
                    </div>
                  </li>
                </ul>

                <p>You can also try restarting this add-on, the supervisor, or Home Assistant.</p>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('google_dns'); return false;">Try Syncing Again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card google_timeout" id="google_timeout">
          <div class="col s12 m12 l9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">cloud_off</i>
                  <span style="vertical-align: top;">Google Drive Connection Timeout</span>
                </span>
                <p>The connection timed out or was closed unexpectedly while trying to communicate with Google Drive.
                  Snapshots can't be
                  backed up without being able to reach Google Drive reliably. This can happen transiently from time to
                  time and the
                  addon will continue to attempt to sync your snapshots. This problem almost always resolves itself
                  after some time.</p>
                <p>Please ensure:</p>
                <ul class="browser-default">
                  <li>The internet is working.</li>
                  <li>The server www.googleapis.com are reachable from your Home Assistant
                    machine. Some ad-blocking and
                    firewall software (eg Ad Guard, pi-hole, etc) can block or throttle traffic to this address.<br />
                    <div style='padding: 5px'>
                      <a onclick="toggleLinkSlide(this, 'cant_reach_google_dns_timeout');" href="#!"><i
                          class="material-icons"
                          style="padding-right: 10px; vertical-align: bottom;">arrow_drop_down</i> Show DNS info</a>
                      <div id="cant_reach_google_dns_timeout" class="default-hidden" style="padding-left: 10px">
                        <p>Below is the list of names and IP Addresses Google's API servers resolve to. Next to each is
                          the result from an attempt the add-on has made to ping the address. This may help in
                          determining if something in your network is blocking accesss to one of Google's API servers,
                          or if Google's API servers are having problems.</p>
                        <div class="dns_info row"></div>
                        <div class="divider"></div>
                      </div>
                    </div>
                  </li>
                </ul>
                <p>You can also try syncing again or restarting this add-on, the supervisor, or Home Assistant.</p>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('google_timeout'); return false;">Try Syncing Again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card google_cant_connect">
          <div class="col s12 m12 l9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">cloud_off</i>
                  <span style="vertical-align: top;">Can't Connect to Google Drive</span>
                </span>
                <p>Google Drive refused an attempt to connect with it. Snapshots can't be backed up without being able
                  to reach Google Drive.</p>
                <p>Please ensure:</p>
                <ul class="browser-default">
                  <li>The internet is working.</li>
                  <li>The server www.googleapis.com are reachable from your Home Assistant
                    machine. Some ad-blocking and
                    firewall software (eg Ad Guard, pi-hole, etc) can block or throttle traffic to this address.<br />
                    <div style='padding: 5px'>
                      <a onclick="toggleLinkSlide(this, 'cant_connect_google_dns');" href="#!"><i class="material-icons"
                          style="padding-right: 10px; vertical-align: bottom;">arrow_drop_down</i> Show DNS info</a>
                      <div id="cant_connect_google_dns" class="default-hidden" style="padding-left: 10px">
                        <p>Below is the list of names and IP Addresses Google's API servers resolve to. Next to each is
                          the result from an attempt the add-on has made to ping the address. This may help in
                          determining if something in your network is blocking accesss to one of Google's API servers,
                          or if Google's API servers are having problems.</p>
                        <div class="dns_info row"></div>
                        <div class="divider"></div>
                      </div>
                    </div>
                  </li>
                </ul>
                <p>You can also try syncing again or restarting this add-on, the supervisor, or Home Assistant.</p>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('google_cant_connect'); return false;">Try Syncing Again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card password_key_invalid">
          <div class="col s12 m12 l9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">vpn_key</i>
                  <span style="vertical-align: top;">Invalid Snapshot Password</span>
                </span>
                <p>The add-on couldn't find your snapshot password in your secrets.yaml file. Snapshots can't be created
                  until you either configure a valid password or remove it.</p>
                <p>This can happen if:</p>
                <ul class="browser-default">
                  <li>Your secrets.yaml file isn't present in your Home Assistant configuration directory</li>
                  <li>The password key isn't present in your secrets.yaml file.</li>
                </ul>
                <p>Please review your <a href="#!" onclick="loadSettings()"><i class="material-icons"
                      style="display: inline; margin-right: 2px; vertical-align: middle; font-size: 15px">settings</i>add-on
                    settings</a>to ensure you've set the snapshot password you intended, then sync again.</p>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('password_key_invalid'); return false;">Try Syncing Again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card snapshot_in_progress">
          <div class="col s12 m12 l9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">settings_backup_restore</i>
                  <span style="vertical-align: top;">Snapshot Already in Progress</span>
                </span>
                <p>The add-on couldn't create a snapshot on schedule because one is already in progress. Home Assistant can
                  only create one snapshot at a time. This can happen if:</p>
                <ul class="browser-default">
                  <li>The add-on was started while a snapshot was already in progress</li>
                  <li>A snapshot was started outside of the add-on, eg manually from the supervisor snapshots page.</li>
                </ul>
                <p>Typically you don't need to do anything to resolve this, as the add-on will pick up again once the
                  snapshot in progress finishes.
                  If you are certain no snapshot is in progress already or this error sticks around, you can try
                  restarting the addon or Home Assistant's host machine.</p>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('snapshot_in_progress'); return false;">Try Syncing Again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card multiple_deletes" id="google_timeout">
          <div class="col s12 m12 l9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">delete_forever</i>
                  <span style="vertical-align: top;">Attempting to Delete Multiple Snapshots</span>
                </span>
                <p>The add-on attempted to delete <span id="data_GoogleDrive">(?)</span> snapshot(s) from Google
                  Drive and <span id="data_HomeAssistant">(?)</span> snapshot(s) from Home Assistant.
                  Under normal circumstance, the add-on will only delete one snapshot at a time (the oldest).
                  This warning is produced to prevent the add-on from accidentally deleting your history of snapshots.
                  This is normal if you recently lowered the number of snapshots to keep in Google Drive or Home
                  Assistant.</p>
                <p>To resolve this you can:</p>
                <ul class="browser-default">
                  <li>Review the <a href="#!" onclick="loadSettings()"><i class="material-icons"
                        style="display: inline; margin-right: 2px; vertical-align: middle; font-size: 15px">settings</i>add-on
                      settings</a> and increase the number of snapshots you keep.</li>
                  <li>Click "Allow Deletion" below to permit these deletions to happen this one time.</li>
                  <li>Click "Never ask again" below to permit the deletes this time and in the future (not recommended)
                  </li>
                </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="allowDeletion('false'); return false;">Allow Deletion</a>
                  <a href="#" onclick="allowDeletion('true'); return false;">Never Ask again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card google_server_error" id="google_server_error">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">cloud_off</i>
                  <span style="vertical-align: top;">Google Server Error</span>
                </span>
                <p>Google returned a server error (HTTP code 500) while trying to backup your snapshots.</p>
                <p>This happens sometimes for reasons beyond anyone's control and almost always the problem goes away on
                  its own. The add-on will attempt uploading again soon, but you can also click "Try Syncing Again"
                  below to make it retry now.</p>
                <p>If this error keeps sticking around, you can try:</p>
                <ul class="browser-default">
                  <li>Making sure the internet is working fine.</li>
                  <li>Making sure you're able to log into your <a href="https://drive.google.com" target="_blank" rel="noreferrer">Google Drive
                      account</a> and upload files without problems.</li>
                  <li>Reauthenting the add-on with Google Drive (click "ACTIONS" -> "Reauthorize Google Drive" up top)
                  </li>
                  <li>Restarting the add-on, Home Assistant, or the supervisor.</li>
                  <li>Reinstalling the add-on.</li>
                </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('google_server_error'); return false;">Try Syncing Again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card google_session_expired" id="google_session_expired">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">cloud_off</i>
                  <span style="vertical-align: top;">Google Session Expired</span>
                </span>
                <p>Google returned a server error (HTTP code 404) while trying to upload your snapshots.</p>
                <p>In most cases, this indicates that your upload took too long to upload. According to Google's
                  documentation if an upload takes longer than a week then it will be forcibly cancelled and return this
                  error. This can happen if your internet is very slow or your snapshots are very large, however the 
                  developer of this addon suspects this can happen in more common circumstances. If the below options don't
                  help, please file an issue on <a onclick="bugReport()">GitHub</a> with as much
                  detail as you can stomach to help him look into it.
                </p>
                <p>If this error keeps sticking around, you can try:</p>
                <ul class="browser-default">
                  <li>Syncing again (if it hasn't already started) by clicking "Try Syncing Again" below.</li>
                  <li>Making sure you're able to log into your <a target="_blank" rel="noreferrer" href="https://drive.google.com">Google
                      Drive
                      account</a> and upload files without problems.</li>
                  <li>Ensuring your internet connection is reasonably fast by doing a <a target="_blank" rel="noreferrer"
                      href="https://www.google.com/search?q=speed+test">speed test</a></li>
                  <li>Making your snapshots smaller. Usually snapshots are large because of a huge sensor database. You
                    can keep a smaller history by setting up the <a target="_blank" rel="noreferrer"
                      href="https://www.home-assistant.io/components/recorder/">recorder component</a> purge settings in
                    Home Assistant</li>
                  <li>Restarting the add-on, Home Assistant, or the supervisor.</li>
                  <li>Reinstalling the add-on.</li>
                </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('google_session_expired'); return false;">Try Syncing Again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card existing_backup_folder" id="existing_backup_folder">
          <div class="col s12 m12 l9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">folder_special</i>
                  <span style="vertical-align: top;">Existing Backup Folder Found</span>
                </span>
                <p>The add-on found a snapshot folder named "<b><span id="data_existing_name"></span></b>" in Google
                  Drive that might already have snapshots
                  in it. This is normal if you've used this addon before, but you'll need to decide where snapshots
                  should go before this will start backing up again. <a id="data_existing_url" target="_blank">Click
                    here</a> to see the folder contents in Google Drive.</p>
                <p>To resolve this you can:</p>
                <ul class="browser-default">
                  <li>"Use the folder" and merge in snapshots from this Home Assistant instance. If this
                    instance has newer snapshots, then some of the snapshots in Google Drive might get deleted to make
                    room for these newer ones (this depends on your settings). You probably want this option if you just
                    reinstalled the addon.</li>
                  <li>"Choose a different folder" to store snapshots in, ignoring the existing folder. You probably want
                    his option if you have multiple Home Assistant instances that should all back up to their
                    own folder.</li>
                  <li>Or "create a new folder" to automatically create a folder and ignore the existing one. The new folder will be created
                    at the root of your My Drive.</li>
                </ul>
                <p>Note: If you run multiple instance of Home Assistant with this addon using the same Google
                  Drive account,
                  its recommended that you use a different folder for each instance. Otherwise they'll delete
                  eachother's
                  snapshots!</p>
                <div class="card-action white-text">
                  <a href="#" onclick="resolvefolder('true')">Use the folder</a>
                  <a href="#" onclick="chooseSnapshotFolder()">Choose a different folder</a>
                  <a href="#" onclick="resolvefolder('false')">Create a new folder</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card backup_folder_missing" id="backup_folder_missing">
          <div class="col s12 m12 l9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">folder_special</i>
                  <span style="vertical-align: top;">Please choose a backup folder</span>
                </span>
                <p class="hide-for-custom-creds">You've choosen to backup your snapshots to a folder of your choosing.
                  Click the link below to open a new window and select the folder in Google Drive you'd like to store
                  your snapshot in.</p>
                <p class="hide-for-default-creds">You've choosen to backup your snapshots to a folder of your choosing.
                  Please navigate to the addon settings to specify a folder to store snapshots in.</p>
                <ul class="browser-default">
                  <li>
                    <p>If you'd rather not specify the folder for uploads, please return to your <a href="#!"
                        onclick="loadSettings()"><i class="material-icons"
                          style="display: inline; margin-right: 2px; vertical-align: middle; font-size: 15px">settings</i>add-on
                        settings</a> and change your preference.</p>
                  </li>
                  <li>
                    Your snapshots in Home Assistant will be re-uploaded to the new folder. If you already have them in
                    an existing folder in Google Drive and you'd like to avoid this, please move them in manually before
                    selecting the folder.
                  </li>
                  <li>
                    <p>If you run multiple instances of
                      Home Assistant with this addon using the same Google Drive account, its recommended that you use a
                      different folder
                      for each instance. Otherwise they'll delete eachother's snapshots!</p>
                  </li>
                </ul>
                <p class="hide-for-custom-creds"><b>Note:</b> Google's authentication mechanisms require you to
                  select your folder through an external domain, which is where the link below will take you.
                  <a href="https://github.com/sabeechen/hassio-google-drive-backup/blob/master/hassio-google-drive-backup/AUTHENTICATION.md"
                    target="_blank" rel="noreferrer">See here</a> for details if you'd like to know more about why.</p>
                <div class="card-action white-text">
                  <a href="#" onclick="chooseSnapshotFolder()" class="hide-for-custom-creds">Choose folder</a>
                  <a href="#" onclick="loadSettings()">Open settings</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card backup_folder_inaccessible" id="backup_folder_inaccessible">
          <div class="col s12 m12 l9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">sync_problem</i>
                  <span style="vertical-align: top;">Couldn't use your backup folder</span>
                </span>
                <p>The addon couldn't use the folder you chose for storing snapshots. The addon can't backup your
                  snapshots
                  until you resolve this. This can happen if:
                  <ul class="browser-default">
                  <li>The folder was deleted</li>
                  <li>You lost permissions on the folder</li> 
                  <li>You signed in with drive using different credentials</li>
                  <li>Something else made the folder unreadable</li>
                  </ul>
                </p>
                <p>
                  To resolve this, you can:
                </p>
                <ul class="browser-default">
                  <li class="hide-for-custom-creds">
                    <p>Use the link below to choose a new folder to store snapshots in.</p>
                  </li>
                  <li class="hide-for-default-creds">
                    <p>Open your <a href="#!" onclick="loadSettings()"><i class="material-icons"
                          style="display: inline; margin-right: 2px; vertical-align: middle; font-size: 15px">settings</i>add-on
                        settings</a> and, enter a valid "Snapshot Folder Id".</p>
                  </li>
                  <li>
                    <p>Open your <a href="#!" onclick="loadSettings()"><i class="material-icons"
                          style="display: inline; margin-right: 2px; vertical-align: middle; font-size: 15px">settings</i>add-on
                        settings</a> and deselect "Manually specify the snapshot folder". The addon will automatically
                      create a new folder for you.</p>
                  </li>
                  <li>
                    <p>If the folder still exists, you can restore access to it by the addon by ensuring that your
                      google account has access to it. The folder should be accessible from <a id="data_existing_url"
                      target="_blank" rel="noreferrer">this link</a>. After doing so, click "sync now" below to try syncing again.</p>
                  </li>
                </ul>
                <p class="hide-for-custom-creds"><b>Note:</b> Google's authentication mechanisms require you to select
                  your folder through an
                  external domain, which is where the "choose folder" link below will take you.
                  <a href="https://github.com/sabeechen/hassio-google-drive-backup/blob/master/hassio-google-drive-backup/AUTHENTICATION.md"
                  target="_blank" rel="noreferrer">See here</a> for details if you'd like to know more about why.</p>
                <div class="card-action white-text">
                  <a href="#" onclick="chooseSnapshotFolder()" class="hide-for-custom-creds">Choose new folder</a>
                  <a href="#" onclick="sync('backup_folder_inaccessible')">sync now</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card cancelled" id="cancelled">
          <div class="col s12 m12 l9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">pause</i>
                  <span style="vertical-align: top;">Backup is Paused</span>
                </span>
                <p>Sync was cancelled because you requested it.  The addon will not resume syncing for another hour unless you take action.</p>
                <br />
                <p>
                  Please note:
                </p>
                <ul class="browser-default">
                  <li>
                    <p>A pending snapshot can't be stopped once its started, because its creation is managed by Home Assistant.  Being "Paused" only stops the addon from creating new snapshots and syncing them with Google Drive.</p>
                  </li>
                  <li>
                    <p>You can resume by clicking the link below or just waiting.</p>
                  </li>
                </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('cancelled')">Resume</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card token_refresh_google_error" id="token_refresh_google_error">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">error</i>
                  <span style="vertical-align: top;">Couldn't Refresh Google Drive Credentials</span>
                </span>
                <p>
                  The addon was unable to refresh your credentials with Google Drive.  When it tried, Google said:
                </p>
                <pre>
                  <span id="data_from_google">(?)</span>
                </pre>
                <p>This can happen if:</p>
                <ul class="browser-default">
                  <li>The domain <a target="_blank" href="https://habackup.io">https://habackup.io</a>, which grants your credentials from Google Drive (<a target="_blank" href="https://github.com/sabeechen/hassio-google-drive-backup/blob/master/hassio-google-drive-backup/AUTHENTICATION.md">click here</a> to see why), is having trouble communicating with Google Drive.</li>
                  <li>Google Drive is experiencing a temporary outage.</li>
                  <li>Something else has gone wrong with your Google Drive credentials.</li>
                </ul>
                <p>Usually this problem will resolve on its own, but you can try:
                  <ul class="browser-default">
                    <li>Syncing again with the button below.  The addon will retry on its own periodically.</li>
                    <li><a href="reauthenticate">Re-authenticating</a> this add-on with Google
                      again. Its unlikely this will resolve the problem, but it also won't cause you to lose any of your data.</li>
                    <li>If the problem persists, <a onclick="bugReport()">file an issue on GitHub</a> to let the developer of this addon know about it.</li>
                  </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('token_refresh_google_error')">sync again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card token_refresh_my_error" id="token_refresh_my_error">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">error</i>
                  <span style="vertical-align: top;">Couldn't Refresh Google Drive Credentials</span>
                </span>
                <p>
                  The addon was unable to refresh your credentials with Google Drive.  When it tried, https://habackup.io said:
                </p>
                <pre>
                  <span id="data_reason">(?)</span>
                </pre>
                <p>This can happen if:</p>
                <ul class="browser-default">
                  <li>The domain <a target="_blank" href="https://habackup.io">https://habackup.io</a>, which grants your credentials from Google Drive (<a target="_blank" href="https://github.com/sabeechen/hassio-google-drive-backup/blob/master/hassio-google-drive-backup/AUTHENTICATION.md">click here</a> to see why), is having some unexpected problems.</li>
                  <li>Something else has gone wrong with your Google Drive credentials.</li>
                </ul>
                <p>Usually this problem will resolve on its own, but you can try:
                  <ul class="browser-default">
                    <li>Syncing again with the button below.  The addon will retry on its own periodically.</li>
                    <li><a href="reauthenticate">Re-authenticating</a> this add-on with Google
                      again. Its unlikely this will resolve the problem, but it also won't cause you to lose any of your data.</li>
                    <li>If the problem persists, <a onclick="bugReport()">file an issue on GitHub</a> to let the developer of this addon know about it.</li>
                  </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('token_refresh_my_error')">sync again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card supervisor_connection" id="supervisor_connection">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">error</i>
                  <span style="vertical-align: top;">Unable to Reach Home Assistant Supervisor</span>
                </span>
                <p>
                  The addon was unable to connect to the Home Assistant Supervisor.  This addon requires the supervisor to be running to create 
                  snapshots, load configuration, etc.  Without the supervisor the addon will not be able to do anything useful.  Typically this 
                  should never happen, and indicates that something is wrong with your Home Assistant installation.
                </p>
                <p>To resolve this you can try:
                  <ul class="browser-default">
                    <li>Syncing again with the button below.  The addon will retry on its own periodically.</li>
                    <li>Restarting the machine that Home Assistant runs on.  Often this resolves transient problems the Home Assistant Supervisor might have.</li>
                    <li>If the problem persists, <a onclick="bugReport()">file an issue on GitHub</a> to let the developer of this addon know about it.</li>
                  </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('supervisor_connection')">sync again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card supervisor_permission" id="supervisor_permission">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">error</i>
                  <span style="vertical-align: top;">Unable to Talk to the Home Assistant Supervisor</span>
                </span>
                <p>
                  The addon was unable to communicate with the Supervisor because it said the addon is unauthorized.  This addon 
                  must communicate with the supervisor to create snapshots, load configuration, etc.  Without the supervisor the 
                  addon will not be able to do anything useful.  Typically this should never happen, and indicates that something
                  is wrong with your Home Assistant installation.
                  <ul class="browser-default">
                    <li>Syncing again with the button below.  The addon will also retry on its own periodically.</li>
                    <li>Restarting the addon, the supervisor, or the whole machine Home Assistant is running on.  Sometimes the supervisor
                      can get into a bad state and this works around it.</li>
                    <li>Checking the Supervisor's logs ("Supervisor" > "System" > "Logs") to see if any errors might indicate what the problem is.</li>
                    <li>If the problem persists, <a onclick="bugReport()">file an issue on GitHub</a> to let the developer of this addon know about it.  He might be able to help.</li>
                  </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('supervisor_connection')">sync again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card supervisor_timeout" id="supervisor_timeout">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">error</i>
                  <span style="vertical-align: top;">A Request to the Supervisor Timed Out</span>
                </span>
                <p>
                  The addon was unable to communicate with the Supervisor because it timed out.  This addon 
                  must communicate with the supervisor to create snapshots, load configuration, etc.  Without the supervisor the 
                  addon will not be able to do anything useful. This usually resolves itself after the addon retrys, but if 
                  you see this error stick around for a while you can try
                  <ul class="browser-default">
                    <li>Syncing again with the button below.  The addon will also retry on its own periodically.</li>
                    <li>Restarting the addon, the supervisor, or the whole machine Home Assistant is running on.  Sometimes the supervisor
                      can get into a bad state and this works around it.</li>
                    <li>If the problem persists, <a onclick="bugReport()">file an issue on GitHub</a> to let the developer of this addon know about it.  He might be able to help.</li>
                  </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('supervisor_timeout')">sync again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card supervisor_unexpected" id="supervisor_unexpected">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">error</i>
                  <span style="vertical-align: top;">The Supervisor Returned an Unexpected Response</span>
                </span>
                <p>
                  The addon was unable to communicate with the Supervisor because it gave back an unreadable response.  
                  This addon must communicate with the supervisor to create snapshots, load configuration, etc.  Without 
                  the supervisor the addon will not be able to do anything useful. This usually 
                  resolves itself after the addon retrys, but if you see this error stick around for a while you can try:
                  <ul class="browser-default">
                    <li>Syncing again with the button below.  The addon will also retry on its own periodically.</li>
                    <li>Restarting the addon, the supervisor, or the whole machine Home Assistant is running on.  Sometimes the supervisor
                      can get into a bad state and this works around it.</li>
                    <li>If the problem persists, <a onclick="bugReport()">file an issue on GitHub</a> to let the developer of this addon know about it.  He might be able to help.</li>
                  </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('supervisor_unexpected')">sync again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card google_unexpected" id="google_unexpected">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">error</i>
                  <span style="vertical-align: top;">Google drive Returned an Unexpected Response</span>
                </span>
                <p>
                  The addon was unable to communicate with Google Drive because it gave back an unreadable response.  
                  Withotu beign able to communicate with google drive, the addon will nto be able to back up snapshots to the cloud.
                  This usually resolves itself after the addon retrys, but if you see this error stick around for a while you can try:
                  <ul class="browser-default">
                    <li>Syncing again with the button below.  The addon will also retry on its own periodically.</li>
                    <li>Restarting the addon, the supervisor, or the whole machine Home Assistant is running on.</li>
                    <li>If the problem persists, <a onclick="bugReport()">file an issue on GitHub</a> to let the developer of this addon know about it.  He might be able to help.</li>
                  </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('supervisor_unexpected')">sync again</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden error_card log_in_to_drive" id="log_in_to_drive">
          <div class="col s12 m12 l9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">
                  <i class="material-icons" style="font-size: 50px; padding-right: 10px;">account_circle</i>
                  <span style="vertical-align: top;">Please log in to Google Drive Manually</span>
                </span>
                <p>The addon couldn't backup your snapshots because it looks like it you've never logged in 
                  to the <a href="https://drive.google.com" target="_blank">Google Drive website</a> before.  Doing this at 
                  least once is required for applications like this to use Google Drive.
                </p>
                <p>
                  To resolve this, you can:
                </p>
                <ul class="browser-default">
                  <li>
                    <p>Visit <a href="https://drive.google.com" target="_blank">https://drive.google.com</a> and log-in with 
                      the account you used to connect this addon to Google Drive, then click "SYNC NOW" below to start syncing.</p>
                  </li>
                  <li >
                    <p><a href="reauthenticate">Reauthorize</a> the addon with Google Drive using a different account.</p>
                  </li>
                </ul>
                <div class="card-action white-text">
                  <a href="#" onclick="sync('backup_folder_inaccessible')">sync now</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden" id="error_details_card">
          <div class="col s12 m9">
            <div class="card red darken-4">
              <div class="card-content white-text">
                <span class="card-title">Error details</span>
                <p id="error_details_paragraph" style="white-space: pre-wrap;font-family: monospace;"></p>
                <div class="card-action white-text">
                  <a target="_blank" onClick="$('#error_details_card').fadeOut(500)">Close</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden" id="error_reports_card">
          <div class="col s12 m9">
            <div class="card">
              <div class="card-content">
                <span class="card-title">Share Error Reports?</span>
                <p>If you select yes the text of unexpected exceptions this add-on produces will be sent to a server
                  maintained by the developer. This is completely optional, but helps give him a heads up for problems
                  with new releases and aids in providing better help messages when things go wrong. If you're unsure,
                  just select "No". You can always change this in the settings menu up on the top right of the page.</p>
                <div class="card-action white-text">
                  <a target="_blank" class="waves-effect waves-light btn" onClick="errorReports('true');">Yes</a>
                  <a target="_blank" class="waves-effect waves-light btn" onClick="errorReports('false');">No</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden" id="ingress_upgrade_card">
          <div class="col s12 m9">
            <div class="card">
              <div class="card-content">
                <span class="card-title">Use Ingress instead?</span>
                <p>The latest version of this add-on supports <a target="_blank" rel="noreferrer"
                    href="https://www.home-assistant.io/blog/2019/04/15/hassio-ingress/">ingress</a>. Ingress allows
                  this web interface to be embeeded inside of Home Assistant, which is usually more secure because it
                  delegates authentication to the same mechanisms you've configured in Home Assistant (eg SSL and
                  username/password). Currently this interface is being served on an additional port outside of
                  Home Assistant which is less secure.</p>
                <br />
                <p><b>Select YES</b> to only use ingress to access this interface. This is recommended, as ingress is
                  almost always more secure.
                  In the future you can access this interface by following the "WEB UI" link from the add-on
                  installation page or embedding it in
                  your Home Assistant side-bar from the same place.</p>
                <br />
                <p><b>Select NO</b> to continue exposing a web interface on an open port with the SSL and authentication
                  settings you choose. The port, which defaults to 1627, can be changed form the add-on page in Home
                  Assistant.</p>
                <br />
                <p>You can change this option any time from the settings menu. No matter which you choose, the interface
                  will be available through ingress.</p>
                <br />
                <div class="card-action white-text">
                  <a target="_blank" class="waves-effect ha-blue waves-light btn"
                    onClick="exposeServer('false');">Yes</a>
                  <a target="_blank" class="waves-effect ha-blue waves-light btn" onClick="exposeServer('true');">No</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row default-hidden" id="restore_help_card">
          <div class="col s12 m9">
            <div class="card">
              <div class="card-content">
                <span class="card-title">Restore a Snapshot</span>
                <p>Snapshots must be restored through the supervisor Snapshots page. You can restore a snapshot by
                  navigating to
                  "Supervisor" &gt; "Snapshots" from the Home Assistant Web-UI,
                  or by clicking the button below to navigate there by a direct link. Depending on how you access the Home
                  Assistant Web-UI the direct link may not work, so you'll just have to navigate there.</p>
                <br />
                <div class="card-action">
                  <a target="_blank" id="restore_hard_link" class="waves-effect ha-blue waves-light btn" href="">Open
                    Snapshots</a>
                  <a target="_blank" class="waves-effect ha-blue waves-light btn modal-close"
                    onClick="$('#restore_help_card').fadeOut(500)">Dismiss</a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="row">
          <div class="col m3">
            <div class="left container" style="margin-left: 15px">
              <h6>In Home Assistant:</h6>
            </div>
            <div class="left container ha_retain_label" style="margin-left: 15px">
              <h7 class="left"><span class="ha_retain_count"></span> saved indefinitely</h7>
            </div>
            <div class="right container">
              <div class="right">
                <h5 id="ha_snapshots" style="text-align: right;"></h5>
                <div id="free_space" style="text-align: right;" class="tooltipped"
                  data-tooltip="An estimate of the space available in your Home Assistant backup folder."></div>
              </div>
            </div>
            <div class="left container" style="margin-left: 15px">
              <h6>In Google Drive:</h6>
            </div>
            <div class="left container drive_retain_label" style="margin-left: 15px">
              <h7 class="left"><span class="drive_retain_count"></span> saved indefinitely</h7>
            </div>
            <div class="right container">
              <h5 class="right " id="drive_snapshots"></h5>
            </div>
            <div class="left container" style="margin-left: 15px">
              <h6>Last Snapshot:</h6>
            </div>
            <div class="right container">
              <h6 class="right">
                <time datetime="" title="" id="last_snapshot"></time>
              </h6>
            </div>
            <div class="left container" style="margin-left: 15px">
              <h6>Next Snapshot:</h6>
            </div>
            <div class="right container">
              <h6 class="right">
                <time datetime="" title="" id="next_snapshot"></time>
              </h6>
            </div>
          </div>
          <div class="col m1 hide-on-med-and-down"></div>
          <div class="col m8">
            <div>
              <div class='row default-hidden' id="no_snapshots_block">
                <div class='col s1'></div>
                <div class='col s2 valign-wrapper center'>
                  <i class='red-text text-darken-4 material-icons' style="font-size: 90px">block</i>
                </div>
                <div class='col s7 center valign-wrapper'>
                  <p>You don't have any snapshots in Google Drive or Home Assistant. Use the "Actions" menu up top
                    create a snapshot and start backing up.</p>
                </div>
              </div>
              <div id="snapshots_loading" class=row>
                <div class='col s1'></div>
                <div class='col s2 valign-wrapper center'>
                  <i class='material-icons' style="font-size: 90px">refresh</i>
                </div>
                <div class='col s7 center valign-wrapper'>
                  <p>Waiting for snapshots to load. If you don't see this go away soon, make sure the add-on is still
                    running. Info on this page may be stale until then.</p>
                </div>
              </div>
              <div id="snapshots">
              </div>
            </div>
            <div class="divider"></div>
          </div>
        </div>
      </div>
    </div>
  </main>
{% endblock %}
